### PR TITLE
MTTL-376 Setup boiler plate code for logging the correlation ID

### PR DESCRIPTION
**Pull request by evak6979**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-376

## Describe this PR

PR to add correlationId for every incoming request.

### *What is the problem we're trying to solve*

CorrelationId will be used as part of our logging strategy. By including a CorrelationId with logs we will be able to trace individual requests during debugging/monitoring attempts.

### *What changes have we introduced*

- Added a correlationId Middleware
- Initialized it in startup
- Added a constants file
- Added unit tests

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

Are there any next steps that need to addressed after merging this PR? Add them here.

### CODE REVIEW

File: BaseApi.Tests/V1/Controllers/BaseControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+using BaseApi.V1.Controllers;
+using BaseApi.V1.Infrastructure;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Routing;
+using NUnit.Framework;
+
+namespace BaseApi.Tests.V1.Controllers
```
Comment (humulla) [inline BaseApi.Tests/V1/Controllers/BaseControllerTests.cs:11]: The folder names need to be changed (both `BaseAPI` and `BaseAPI.Tests`)

Comment (evak6979) [reply BaseApi.Tests/V1/Controllers/BaseControllerTests.cs:11]: Happy to do so, but if possible we should merge the initial renaming PR. And then I can merge master from there to this branch which should resolve it.

Comment (humulla) [reply BaseApi.Tests/V1/Controllers/BaseControllerTests.cs:11]: I forgot the renaming PR hasn't been merged, ignore this comment please

Comment (evak6979) [reply BaseApi.Tests/V1/Controllers/BaseControllerTests.cs]: No worries at all :) With 2 approvals, I took the liberty of merging the renaming PR to master and then into this branch. Renaming issues should be resolved by now.

**Review by mikemoody-amido — APPROVED** [review]

**Pull request merged**